design-toolkit/components/src/components/hero/index.tsx

```diff
@@ -0,0 +1,117 @@
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { Icon } from '@/index';
+import type { PropsWithChildren, ReactElement } from 'react';
+import { Children, createContext, isValidElement, useContext } from 'react';
+import type { ContextValue } from 'react-aria-components';
+import { type HeroStyleVariants, HeroStyles } from './styles';
+
+interface ChildProps extends PropsWithChildren {
+  className?: string;
+}
+
+interface HeroProps extends HeroStyleVariants, PropsWithChildren {
+  className?: string;
+}
+
+export const HeroContext =
+  createContext<ContextValue<HeroProps, HTMLDivElement>>(null);
+
+export function Hero({ children, className, ...props }: HeroProps) {
+  const slots = {
+    icon: null as ReactElement | null,
+    primary: null as ReactElement | null,
+    secondary: [] as ReactElement[],
+    invalid: [] as ReactElement[],
+  };
+  const styles = HeroStyles({ ...props });
+
+  Children.forEach(children, (child) => {
+    if (isValidElement(child)) {
+      const childType = child.type as { displayName?: string };
+      const displayName = childType?.displayName;
+
+      if (displayName === 'Hero.Icon') {
+        slots.icon = child;
+      } else if (displayName === 'Hero.Primary') {
+        slots.primary = child;
+      } else if (displayName === 'Hero.Secondary') {
+        slots.secondary.push(child);
+      } else {
+        slots.invalid.push(child);
+      }
+    }
+  });
+
+  return (
+    <HeroContext.Provider value={props}>
+      <header className={styles.container({ className })}>
+        <aside>{slots.icon}</aside>
+        <main>
+          {slots.primary}
+          {slots.secondary}
+
+          {slots.invalid.length > 0 && (
+            <div className='mt-l rounded-large bg-[hotPink] p-l'>
+              <h1 className='text-header-l'>Invalid {'<Hero />'} content</h1>
+              {slots.invalid}
+            </div>
+          )}
+        </main>
+      </header>
+    </HeroContext.Provider>
+  );
+}
+
+Hero.Icon = HeroIcon;
+Hero.Primary = HeroPrimary;
+Hero.Secondary = HeroSecondary;
+
+function HeroIcon({ children, className }: ChildProps) {
+  const { compact, theme } = useHeroVariants();
+  const styles = HeroStyles({ compact, theme });
+
+  return <Icon className={styles.icon({ className })}>{children}</Icon>;
+}
+HeroIcon.displayName = 'Hero.Icon';
+
+function HeroPrimary({ children, className }: ChildProps) {
+  const { compact, theme } = useHeroVariants();
+  const styles = HeroStyles({ compact, theme });
+
+  return <div className={styles.primary({ className })}>{children}</div>;
```

Comment:
If you use Heading and HeadingContext, then you'd be able to get rid of these components as well. You'd use the Provider to supply the contexts

```
<Provider values={[
    [IconContext, { className: icon() }],
    [HeadingContext, {
       [DEFAULT_SLOT]: { className: heading(), level: 2 },
       subtitle: { className: subtitle() }
    }
]]}>
  <Header>{children}</Header>
</Provider>
```

```
<Hero>
  <Icon><Placeholder /></Icon>
  <Heading>Foobar</Heading>
  <Heading slot="subtitle">The questionable</Heading>
</Hero>
```

I want to get us out of the habit of implementing tiny wrapper components that do nothing but apply a className

Reply:
Even better, use Text in place of the secondary/subtitle heading